src/mob.rs

```diff
@@ -0,0 +1,87 @@
+use bevy::app::{App, Plugin, Update};
+use bevy::math::{Vec2, Vec3};
+use bevy::prelude::{
+    default, Color, Commands, Component, Query, Res, ResMut, Resource, Sprite, SpriteBundle, Time,
+    Timer, Transform, With,
+};
+use bevy::time::TimerMode;
+use bevy::window::{PrimaryWindow, Window};
+use rand::Rng;
+
+static MOB_SPAWN_TIME: f32 = 5.;
+
+pub struct MobPlugin;
+impl Plugin for MobPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Update, move_mob);
+        app.add_systems(Update, spawn_mobs_over_time);
+        app.init_resource::<MobSpawnTimer>();
+        app.add_systems(Update, tick_mob_spawn_timer);
+    }
+}
+
+#[derive(Resource)]
+pub struct MobSpawnTimer {
+    pub timer: Timer,
+}
+
+impl Default for MobSpawnTimer {
+    fn default() -> MobSpawnTimer {
+        MobSpawnTimer {
+            timer: Timer::from_seconds(MOB_SPAWN_TIME, TimerMode::Repeating),
+        }
+    }
+}
+
+pub fn tick_mob_spawn_timer(mut mob_spawn_timer: ResMut<MobSpawnTimer>, time: Res<Time>) {
+    mob_spawn_timer.timer.tick(time.delta());
+}
+
+pub fn spawn_mobs_over_time(
+    mut commands: Commands,
+    window_query: Query<&Window, With<PrimaryWindow>>,
+    mob_spawn_timer: Res<MobSpawnTimer>,
+) {
+    if mob_spawn_timer.timer.finished() {
```

Comment:
for better code readability this should be inverted and handled via an early return 

here's a [video](https://www.youtube.com/watch?v=CFRhGnuXG-4) which explains the reasons behind this